### PR TITLE
feature: add tests

### PR DESCRIPTION
**Pull request by ariadnafer**
# Description

Added testing with coverage
Added tests for the following components:
- extract_bq_to_dataset
- custom_train_job
- import_model_evaluation

# How has this been tested?

This PR is for testing purposes itself

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [x] I have assigned a reviewer and messaged them

# Pipeline run links:

### CODE REVIEW

File: components/bigquery-components/tests/test_extract_bq_to_dataset.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import bigquery_components
+
+extract_bq_to_dataset = bigquery_components.extract_bq_to_dataset.python_func
+
+
+def test_extract_bq_to_dataset(tmpdir):
+    with mock.patch("google.cloud.bigquery.client.Client") as mock_client, mock.patch(
+        "google.cloud.bigquery.job.ExtractJobConfig"
+    ) as mock_job_config, mock.patch("google.cloud.bigquery.table.Table") as mock_table:
```
Comment (felix-datatonic) [inline components/bigquery-components/tests/test_extract_bq_to_dataset.py:13]:
```suggestion
@mock.patch("google.cloud.bigquery.client.Client")
@mock.patch("google.cloud.bigquery.job.ExtractJobConfig")
@mock.patch("google.cloud.bigquery.table.Table")
def test_extract_bq_to_dataset(tmpdir, mock_client, mock_job_config, mock_table):
```

Comment (felix-datatonic) [reply components/bigquery-components/tests/test_extract_bq_to_dataset.py:13]: I find this more readable, thoughts?

**Review by felix-datatonic — COMMENTED** [review]
Thanks @ariadnafer! Have left a minor comment on formatting `mock.patch`, otherwise happy with improving the code coverage. Also, I'd remove inline comments as much as possible and only comment complex code blocks. Python code should aim for simplicity and self-explainability.

File: components/bigquery-components/tests/test_extract_bq_to_dataset.py (hunk 1 above)
Comment (ariadnafer) [reply components/bigquery-components/tests/test_extract_bq_to_dataset.py:13]: I agree, I prefer that way but I just wanted to be consistent with what it was already there. I will implement it in all the tests of the project :)

**Review by felix-datatonic — APPROVED** [review]

**Pull request merged**